### PR TITLE
🎉 Source Google Analytics v4: Declare oauth parameters in google sources

### PR DESCRIPTION
**Pull request by gaart**
## What
Add support for connecting via Ouath webflow

## How
Update spec.json schema to support 2 types of authorization
Update source.py
Update unit_tests

#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [x] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json [hunk 1]
```diff
@@ -4,49 +4,9 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "title": "Google Analytics V4 Spec",
     "type": "object",
-    "required": ["credentials", "view_id", "start_date"],
-    "additionalProperties": true,
+    "required": ["authorization", "view_id", "start_date"],
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json:7]: we should keep the name the same since [other code](https://github.com/airbytehq/airbyte/blob/master/airbyte-oauth/src/main/java/io/airbyte/oauth/flows/google/GoogleAnalyticsOAuthFlow.java#L57) in the platform depends on this

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json [hunk 2]
```diff
@@ -69,17 +29,67 @@
...
+                  "{ \"type\": \"service_account\", \"project_id\": YOUR_PROJECT_ID, \"private_key_id\": YOUR_PRIVATE_KEY, ... }"
+                ]
+              }
+            }
+          }
+        ]
       }
     }
-  },
-  "authSpecification": {
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json:75]: this should remain, why remove it?

**Review by keu — APPROVED** [review]

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py [hunk 3]
```diff
@@ -469,17 +442,34 @@ def get_refresh_request_params(self) -> Mapping[str, any]:
         }
         headers = {"kid": self.client_id}
         signed_jwt = jwt.encode(payload, self.client_secret, headers=headers, algorithm="RS256")
-        return {"grant_type": "urn:ietf:params:oauth:grant-type:jwt-bearer", "assertion": signed_jwt}
+        return {"grant_type": "urn:ietf:params:oauth:grant-type:jwt-bearer", "assertion": str(signed_jwt)}
```
Comment (ChristopheDuong) [inline airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py:445]: Since line 432 was removed, it doesn't seem to be using the method from super() in non-jwt cases... (ie when not using service account json?)

In those cases, it seems the grant_type should be simply a `refresh_token` value:
https://github.com/airbytehq/airbyte/blob/f88b8313a81da3c30a6daefc26aaacfad8332b5a/airbyte-cdk/python/airbyte_cdk/sources/streams/http/auth/oauth.py#L59

Is this making the connector supporting only the service account / jwt?

Comment (gaart) [reply airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py:445]: The get_authenticator method contains the logic for choosing the authorization type, there is support for client and service authorization

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json [hunk 4]
```diff
@@ -4,50 +4,9 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "title": "Google Analytics V4 Spec",
     "type": "object",
-    "required": ["credentials", "view_id", "start_date"],
+    "required": ["view_id", "start_date"],
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json:7]: why remove credentials as required?

File: airbyte-integrations/connectors/source-google-analytics-v4/unit_tests/unit_test.py [hunk 5]
```diff
@@ -81,14 +84,14 @@ def test_check_connection_oauth(jwt_encode_mock, mocker, mock_metrics_dimensions
...
     del test_config["custom_reports"]
     test_config["credentials"] = {
+        "auth_type": "Client",
         "client_id": "client_id_val",
         "client_secret": "client_secret_val",
         "refresh_token": "refresh_token_val",
     }
     source = SourceGoogleAnalyticsV4()
     assert source.check_connection(MagicMock(), test_config) == (True, None)
     jwt_encode_mock.encode.assert_not_called()
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-google-analytics-v4/unit_tests/unit_test.py:94]: why was this removed?

**Review by sherifnada — APPROVED** [review]
